### PR TITLE
Fix: Safe chainid for metamask (#4)

### PR DESCRIPTION
**Pull request by sambacha**
Metamask has a hardcoded value for rejecting chainId's over a certain number. That number is calculated based off of 

[source from remarks, metamask engineer](https://gist.github.com/rekmarks/a47bd5f2525936c4b8eee31a16345553)
```
From ethereumjs-util@7.0.5, we have that:

  v = recovery + (chainId * 2 + 35)

Per the above discussion, we also have that:

  int_max = 2**53 - 1
  recovery_max = 3
  chainId_max = ?

Therefore:

  v_max = 3 + (chainId * 2 + 35) = chainId * 2 + 38
    &&
  v_max <= int_max
  
    ->
    
  2**53 - 1 = MAX_SAFE_CHAIN_ID * 2 + 38
  
    ->
    
  // Since we're dealing with integers, we round down.
  
  MAX_SAFE_CHAIN_ID = floor( ( 2**53 - 39 ) / 2 ) = 4503599627370476
```

Ethereumjs-util as referenced above has BigInt support since 7.0.9, regardless metamask has encoded this value within their application.
[see https://github.com/ethereumjs/ethereumjs-util/releases/tag/v7.0.9](https://github.com/ethereumjs/ethereumjs-util/releases/tag/v7.0.9)

I tried to document it as best I could, I don't know how you feel about that I can remove it if you would like.

### CODE REVIEW

**Comment by sambacha** [comment]
@NoahZinsmeister if you can review ser 🙏

**Comment by sambacha** [comment]
@talentlessguy or @NoahZinsmeister

**Comment by NoahZinsmeister** [comment]
hey @sambacha sorry for the delay! happy to add this, but migrating development efforts to v8 (in the `main` branch), would you mind adding this check there instead? relevant line is https://github.com/NoahZinsmeister/web3-react/blob/main/packages/store/src/index.ts#L6

**Comment by sambacha** [comment]
> hey @sambacha sorry for the delay! happy to add this, but migrating development efforts to v8 (in the `main` branch), would you mind adding this check there instead? relevant line is [`main`/packages/store/src/index.ts#L6](https://github.com/NoahZinsmeister/web3-react/blob/main/packages/store/src/index.ts?rgh-link-date=2022-01-11T18%3A36%3A23Z#L6)

Hey @NoahZinsmeister - thanks for touching base, will update with your feedback. Appreciate the followup  - cheers!

File: packages/store/src/index.ts [hunk 1]
```diff
@@ -11,9 +11,8 @@ import create from 'zustand/vanilla'
  * ```
  *
  * @const MAX_SAFE_CHAIN_ID The maximum integer value that metamask will accept as a chainId
- * @see {@link https://git.io/JPBat}
+ * @see {@link https://github.com/MetaMask/metamask-extension/blob/b6673731e2367e119a5fee9a454dd40bd4968948/shared/constants/network.js#L31}
```
Comment (sambacha) [inline packages/store/src/index.ts:14]: 👍   - didn't know how strict your style guide was for line limit, thank you

**Comment by sambacha** [comment]
thanks @NoahZinsmeister

**Pull request merged**